### ISSUE TITLE
Regression with cursor height when using custom `line_height` config?

### ISSUE THREAD

**Issue by jesseleite** [wez/wezterm#4591]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20231112-171650-721fbdf5

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I updated Wezterm nightly (it has been months since I updated), and now the cursor height doesn't match my configured `line_height` anymore. Could this be a regression?

![CleanShot 2023-11-17 at 13 28 22](https://github.com/wez/wezterm/assets/5187394/5839748e-b2ca-4e3b-84d1-699ff8baee5d)

### To Reproduce

Set `line_height = 1.6` and highlight some text.

### Configuration

```lua
line_height = 1.6
```

### Expected Behavior

I would assume cursor height should always match your line_height, as it did in previous Wezterm versions?

### Anything else?

Nope! Just a polite 'thank you' for all your rad work on WezTerm! ❤️

**Comment by wez** [comment]
See: 
* https://github.com/wez/wezterm/issues/2882

**Comment by ilias777** [comment]
Hi and sorry to write in this closed issue. 
Since the last update the cursor looks for me weird. I don't understand why people don't like to render the cursor at the line height.

Before the update it looks, in my opinion, right:
![cursor1](https://github.com/wez/wezterm/assets/39212564/9ee5030b-2ef1-4eb7-9ce7-03453a13fb09)

After the update the cursor is to small and is not catching with the eye right away:
![cursor2](https://github.com/wez/wezterm/assets/39212564/f3bf55dd-6b93-43c1-a74b-00f29dc1a6b5)

I searched in the docs, in the two issues #4591 and #2882 and in your commt 2c95b98 for any option to change the cursor height, but I didn't find anything.

How can I set the cursor height to fit with the line height?

Thank you very much for any help.
